capplets/appearance/appearance-font.c

```diff
@@ -421,20 +421,40 @@ dpi_from_pixels_and_mm (int pixels, int mm)
 }
 
 static double
-get_dpi_from_x_server (void)
+get_dpi_from_x_server_or_monitor (void)
 {
   GdkScreen  *screen;
+  GdkDisplay *display;
   double dpi;
 
   screen = gdk_screen_get_default ();
+  display = gdk_display_get_default();
 
   if (screen) {
     double width_dpi, height_dpi;
 
-    Screen *xscreen = gdk_x11_screen_get_xscreen (screen);
+    if (GDK_IS_X11_DISPLAY (display))
+    {
+      Screen *xscreen = gdk_x11_screen_get_xscreen (screen);
 
-    width_dpi = dpi_from_pixels_and_mm (WidthOfScreen (xscreen), WidthMMOfScreen (xscreen));
-    height_dpi = dpi_from_pixels_and_mm (HeightOfScreen (xscreen), HeightMMOfScreen (xscreen));
+      width_dpi = dpi_from_pixels_and_mm (WidthOfScreen (xscreen), WidthMMOfScreen (xscreen));
+      height_dpi = dpi_from_pixels_and_mm (HeightOfScreen (xscreen), HeightMMOfScreen (xscreen));
+    }
+    else
+    {
+      GdkMonitor *monitor;
+      GdkRectangle geometry = {0};
+
+      /*FIXME: we have to use the main monitor for this
+       *which may not always be the leftmost monitor
+       *Separate per-monitor settings for this would be ideal
+       */
+      monitor = gdk_display_get_monitor (display, 0);
```

Comment:
Why is it ```0``` here? If multiple monitors are connected, will there be a problem here? I'm not sure

Reply:
We are setting the font here, and right now I have yet to figure out how to do this on a per-monitor basis. First priority on that would be showing the caja desktop on all monitors. Monitor 0 (leftmost or only monitor on single monitor setup) is always present, and in the wayland session it's where caja's desktop goes unless manually started from a terminal on another monitor.

Reply:
Would it be better to use ```gdk_display_get_primary_monitor``` ?

Reply:
Would it be better to use  ```gdk_display_get_monitor_at_window``` instead

Reply:
We don't have the window the capplet is using within `get_dpi_from_x_server_or_monitor`, and we don't have it in the calling function either. Thus we'd have to pass the window through a lot of other functions to get it here it seems.

In an ideal multimonitor case, an argument could be made for using the monitor the capplet is on, or for using the monitor the desktop is on. Setting a different font for each monitor would require a lot of other changes, notable additional gsettings values to store additional values in, we'd have to show each possible monitor in the capplet at well

Reply:
I personally think there's no problem. Let others take a look too